### ISSUE TITLE
When program finished,PrefetchDataZMQ do not exit graceful

### ISSUE THREAD

**Issue by tonyw** [ppwwyyxx/tensorpack#562]
When program finished,I found the process do not exit graceful.
I check the process and the code.I found the reason is that PrefetchDataZMQ 
do not terminate the sub processes successfully.
In code https://github.com/ppwwyyxx/tensorpack/blob/4c02f009e8448d74478a4b61e970d2b54ad0f599/tensorpack/dataflow/prefetch.py#L102
I think there should be check sub processes status,If the sub processes do not terminated,should   kill it directly.

**Comment by ppwwyyxx** [comment]
Is there a reproducible code? Could you provide your environment info?

**Comment by jbescudie** [comment]
Same here. The following code reproduces the error under Ubuntu 16.04, python 2.7, with tensorpack updated from pip today :
```
import tensorpack as tp
ds=tp.DataFromList(range(10))
ds=tp.PrefetchDataZMQ(ds, 6)
tp.TestDataSpeed(ds).start_test()
```

**Comment by ppwwyyxx** [comment]
Interesting. The code can exit correctly on my machine.

**Comment by jbescudie** [comment]
the code is run in an IPython console (in PyCharm), I'm left with 7 python2.7 processes : the original one and the 6 forks. And 6 forks remains after exiting the main process. I tried in a virtualenv and without.

**Comment by jbescudie** [comment]
After further testing, I agree it works as it should: processes are closed after correctly exiting the main process with exit(). It doesn't work with the way pycharm closes the session though, so that is not a tensorpack issue.

So I have a feature request regarding this behaviour: _add a method to manually close processes created by PrefetchData and PrefetchDataZMQ_. It would be really useful during development for those who don't restart the interpreter all the time.

PS: I had an unexpected behaviour: having a file called zmq.py in my path messed up loading tensorpack module. I can live with that, if you want more info, I'll open another issue.

**Comment by ppwwyyxx** [comment]
To manually close processes you can call `ds.__del__()` when ds is `PrefetchData` or `PrefetchDataZMQ`.

Name conflicting is a general Python issue and you should just avoid putting your own code into PYTHONPATH.

**Comment by jbescudie** [comment]
Thanks for your anwser. The ds.__del__() method doesn't close processes when run from command line. I used this code:
`
import tensorpack as tp
ds=tp.DataFromList(range(10))
ds=tp.PrefetchDataZMQ(ds, 6)
tp.TestDataSpeed(ds).start_test()
raw_input("Enter to clean")
ds.__del__()
raw_input("Continue")  # processes are still there 
exit()
`
Although, if run in debug mode with a breakpoint on the 'Continue' line, the processes are gone (and marked as TERMINATED in ds._procs).

For the naming conflict, I meant my file for the first code I submitted in this issue was named zmq.py. Maybe I'm wrong, but I don't think it should mix with another import inside a module.
